packages/shared/src/utils/validation/declarative-validation-rules.ts

```diff
@@ -19,7 +19,12 @@ const validRequired = (value?: string | number /* , options, field */) => {
     const str = value.toString().replace(/\s/g, '');
     return str.length > 0;
 };
-export const validAddress = (value: string) => !/[`~!$%^&*_=+[}{\]\\"?><|]+/.test(value);
+export const address_permitted_special_characters_message_string = ". , ' : ; ( ) ° @ # / -";
+export const validAddress = (value: string) => {
+    const is_ok = /^[\p{L}\p{Nd}\s'.,:;()\u00b0@#/-]{1,70}$/u.test(value);
+    const message = is_ok ? null : form_error_messages.address();
+    return { is_ok, message };
```

Comment:
Btw the idea of returning `{ is_ok: bool, message: string }` is taken from the `validNumber` function which is in the same file, seems like a nice way to encapsulate the error messages.